homeassistant/components/group.py

```diff
@@ -161,48 +169,82 @@ def reload_service_handler(service_call):
     return True
 
 
-def _process_config(hass, config, component):
+@asyncio.coroutine
+def _async_process_config(hass, config, component):
     """Process group configuration."""
+    groups = []
     for object_id, conf in config.get(DOMAIN, {}).items():
         name = conf.get(CONF_NAME, object_id)
         entity_ids = conf.get(CONF_ENTITIES) or []
         icon = conf.get(CONF_ICON)
         view = conf.get(CONF_VIEW)
 
-        group = Group(hass, name, entity_ids, icon=icon, view=view,
-                      object_id=object_id)
-        component.add_entities((group,))
+        # This order is important as groups get a number based on creation
+        # order.
```

Comment:
Is it possible that multiple _async_process_config's may be running at once in the event loop? I'm not sure what number is referred to here, but since yield from is going to let other code on the event loop run before the result comes back, it's possible if other _async_process_config's are running that their async_create_group calls will be interleaved with this one. Hopefully that's not a problem.